### ISSUE TITLE
[4.0] Custom WrapperAttributes overrides JS init in some fields

### ISSUE THREAD

**Issue by patrickleogit** [Laravel-Backpack/CRUD#2188]
# Bug report

### What I did

I've inserted a field of type "Upload" in one of my CRUD pages. I've then added a WrapperAttributes inside of it in order to change the field's classes and its size as a result.

### What I expected to happen

I expected the "Upload" field to run properly with just its size changed.

### What happened

The clear button of the upload field didn't work and the upload field itself never got updated whenever I tried to modify it. Uploading the first time worked (although the label containing the filename didn't get updated). 

### What I've already tried to fix it

I took a look at the original blade file of the upload field. It seems like the function that gets called, the one that updates the field (bpFieldInitUploadElement), doesn't get added if WrapperAttributes is already used by the user for other stuff (as it checks for it to be empty before adding the data-init-function and data-field-name attributes). My fix for now was to add these two in the WrapperAttributes I've added in my CRUD page. It works, but I thought about reporting the issue anyway, since customizing the WrapperAttributes of the upload field can result in it not working properly anymore.

### Backpack, Laravel, PHP, DB version

laravel/framework                         v5.8.35 
backpack/crud                              4.0.12            
backpack/generators                    2.0.4             
backpack/permissionmanager      5.0.2

**Comment by pxpm** [comment]
Looking at this issue, i think the simpler solution would be to create a new field key for advanced use cases where overriding the js functions is necessary. (I don't see it been used that often but .. it's just a field configuration in an advanced scenario, not default one).

```
'js_init' => [
'data-init-function' => 'bpFieldInitUploadMultipleElement',
'data-field-name' => 'extra_photos'
]
```

Keep the same usual stuff `col-md-6 etc` in wrapperAttributes key. If not set the `js_init` crud should load just "similar as default" to other fields. 

@tabacitu let me know your opinion on this one . Or if there are other reasons for this behaviour i'm not seeing. 

Best,
Pedro

**Comment by tabacitu** [comment]
@patrickleogit , @johnpuddephatt - thank you for reporting this!

@pxpm - a new ```js_init``` would indeed look clean when defining a field, but I'm afraid it would modify the fields JS, unnecessarily. 

I think a simpler solution would be to change the condition for adding that particular wrapper_attributes attribute. To rephrase - it's currently:
```php
    if (!isset($field['wrapperAttributes']) || !isset($field['wrapperAttributes']['class']))
    {
        $field['wrapperAttributes']['data-init-function'] = 'bpFieldInitUploadMultipleElement';
        $field['wrapperAttributes']['data-field-name'] = $field['name'];
    }
```

But it should probably be something like:

```php
    if (!isset($field['wrapperAttributes']) || !isset($field['wrapperAttributes']['data-init-function']))
    {
        $field['wrapperAttributes']['data-init-function'] = 'bpFieldInitUploadMultipleElement';
    }
    if (!isset($field['wrapperAttributes']) || !isset($field['wrapperAttributes']['data-field-name']))
    {
        $field['wrapperAttributes']['data-field-name'] = $field['name'];
    }
```

That:
- should be less intrusive (no need to change anything in the field JS)
- should still allow the user to change that particular ```data-field-name``` or ```data-init-function``` if they want;
- should still fix the bug at hand (since people would be able to define ```$field['wrapperAttributes']['class']``` and it would just pick up the default value for ```data-field-name``` or ```data-init-function```;

Let me know what you think.
Cheers!

**Comment by pxpm** [comment]
Sorry. Already submitting the PR while you write. Fortunately I spotted that issue with javascript too an decided for a cleaner way. Let me know what you think in the PR. 

Best, 
Pedro

**Comment by patrickleogit** [comment]
@tabacitu - Yeah, I think changing the condition the way you explained in your last post is the best solution to this issue for the reasons you've listed. 

Thank you!

**Comment by tabacitu** [comment]
Fixed in #2211 by @pxpm 's PR, which is now merged and tagged. Thanks a lot for the help @patrickleogit & @pxpm !
